packages/admin/cms-admin/src/form/file/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,175 @@
+    const [failedToDownloadFileIds, setFailedToDownloadFileIds] = React.useState<string[]>([]);
+
+    const apiUrl = "http://localhost:4000"; // TODO: Where do we get the url from? Env? Hook?
+
+    const singleFile = maxFiles === 1;
+    const inputValue = React.useMemo(() => (Array.isArray(input.value) ? input.value : input.value ? [input.value] : []), [input.value]);
+
+    const files = [...inputValue, ...failedUploads, ...uploadingFiles].map((file) => {
+        if ("id" in file && downloadingFileIds.includes(file.id)) {
+            return { ...file, isDownloading: true };
+        }
+
+        if ("id" in file && failedToDownloadFileIds.includes(file.id)) {
+            return { ...file, error: <FormattedMessage id="comet.finalFormFileUpload.downloadFailed" defaultMessage="Download failed." /> };
+        }
+
+        return file;
+    });
+
+    return (
+        <FileSelect<GQLFinalFormFileUploadFragment>
+            onDrop={async (acceptedFiles, rejectedFiles) => {
+                setFailedUploads([]);
+
+                const tooManyFilesWereDropped = rejectedFiles.some((rejection) => rejection.errors.some((error) => error.code === "too-many-files"));
+                setTooManyFilesSelected(tooManyFilesWereDropped);
+
+                rejectedFiles.map((rejection) => {
+                    const failedFile: ErrorFileSelectItem = {
+                        name: rejection.file.name,
+                        error: true,
+                    };
+
+                    if (rejection.errors.some((error) => error.code === "file-too-large")) {
+                        failedFile.error = <FormattedMessage id="comet.finalFormFileUpload.fileTooLarge" defaultMessage="File is too large." />;
+                    }
+
+                    setFailedUploads((existing) => [...existing, failedFile]);
+                });
+
+                if (singleFile) {
+                    input.onChange(undefined);
+                }
+
+                if (tooManyFilesWereDropped || !acceptedFiles.length) {
+                    return;
+                }
+
+                setUploadingFiles(acceptedFiles.map((file) => ({ name: file.name, loading: true })));
+
+                const fetches = acceptedFiles.map((file) => {
+                    const formData = new FormData();
+                    formData.append("file", file);
+                    return fetch(`${apiUrl}/public-upload/files/upload`, {
+                        method: "POST",
+                        body: formData,
+                    });
+                });
+
+                const responses = await Promise.all(fetches)
+                    .then((responses) => Promise.all(responses.map((response) => response.json() as Promise<ApiResponse>)))
+                    .finally(() => {
+                        setUploadingFiles(uploadingFiles.filter((loadingFile) => !acceptedFiles.some((file) => file.name === loadingFile.name)));
+                    });
```

Comment:
Is it a good idea to do this in parallel? Could this be too much work for the API?

Reply:
I have no idea how resource intensive file uploads are and how many resources our APIs generally have available 🤔 

Do you know what would happen, if the API is overloaded because of this?
Would it crash or just be slow but reliable?

Reply:
> Would it crash or just be slow but reliable?

I think it would crash.

In the DAM we upload sequential, so maybe it's best to do this here as well.

Reply:
The files are now uploaded sequentially